analyzers/dataframe/src/CaloNtupleizer.cc

```diff
@@ -211,6 +211,28 @@ ROOT::VecOps::RVec<int> getCaloCluster_lastCell (const ROOT::VecOps::RVec<edm4he
   return result;
 }
 
+ROOT::VecOps::RVec<std::vector<float>>
+getCaloCluster_energyInLayers (const ROOT::VecOps::RVec<edm4hep::ClusterData>& in,
+                               const ROOT::VecOps::RVec<edm4hep::CalorimeterHitData>& cells) {
+  static const int layer_idx = m_decoder->index("layer");
+  static const int cryo_idx = m_decoder->index("cryo");
+  ROOT::VecOps::RVec<std::vector<float>> result;
+  result.reserve(in.size());
+
+  for (const auto & c: in) {
+    std::vector<float> energies(12, 0); // WARNING Hardcode 12 layers for now. Flemme-%
```

Comment:
One can retrieve the max number of layer with  `m_decoder->operator[](m_decoder->index("layer")).maxValue() - m_decoder->operator[](m_decoder->index("layer")).minValue()` but this gives the maximum foreseen in the segmentation, not the number of layer actually used/implemented (256 in our case since we reserve 8 bits for the layer field). Is this annoying? If yes, could you pass this number as an argument in the python wrapper `examples/FCCee/fullSim/caloNtupleizer/analysis.py`?

Reply:
Ok, parameter `nLayers` added.